lib/services/messenger.js

```diff
@@ -6,9 +6,67 @@ module.exports = messengerServiceFactory;
 
 messengerServiceFactory.$provide = 'Services.Messenger';
 messengerServiceFactory.$inject = [
-    'Messenger'
+    'Messenger',
+    
 ];
 
 function messengerServiceFactory(Messenger) {
-    return new Messenger();
+    function MessengerService(){
+        this.messenger = new Messenger();
+    }
+
+	MessengerService.prototype.validate = function validate(data) {
+		return this.messenger.validate(data);
+	}
+	
+	MessengerService.prototype.queueName = function queueName(name, routingKey){
+		return this.messenger.queueName(name, routingKey);
+	}
+	
+	MessengerService.prototype.start = function start() {
+        return this.messenger.start();
+    };
+
+    MessengerService.prototype.stop = function stop() {
+        return this.messenger.stop();
+    };
+	
+    MessengerService.prototype.publishExternalEvents  = function(name, routingKey, data, options) {
+        assert.string(name, 'name');
+        assert.string(routingKey, 'routingKey');
+        assert.object(data, 'data');
+
+        this.messenger.publish(name, routingKey, data, options);
+    };
+
+	
+    MessengerService.prototype.publishInternalEvents  = function(name, routingKey, data, options) {
+        return this.messenger.publish(name, routingKey, data, options);
+    };
+
+
+    MessengerService.prototype.publish  = function(name, routingKey, data, options) {
+        return this.messenger.publish(name, routingKey, data, options);
+    };
+
+
+    MessengerService.prototype.subscribe  = function(name, routingKey, callback, type) {
+        return this.messenger.subscribe(name, routingKey, callback, type);
+    };
+
+    MessengerService.prototype.request = function(name, routingKey, data, type, timeout) {
+        return this.messenger.request(name, routingKey, data, type, timeout);
+    };
+
+
+	MessengerService.prototype.safeSubscriptionDispose = function (context) {
+		return this.messenger.safeSubscriptionDispose(context);
+	}
+
+   MessengerService.prototype.subscribeCallback = function(context, options) {
+	   return this.messenger.subscribeCallback(context, options);
+   }
+    
+    return new MessengerService();
 }
+
```

Comment:
I thought of inherits at first, but according to @yyscamper the relationship between this messenger.js and the messenger.js in lib/common are not in inherits relationships, being an instance of it is a better choice. As for the webhook, it could be treated in a similar way: in publishExternalEvents, this.hook = new Hook() or something like that.